File: solutions/java/yoosieun/실패율.java
import java.util.*;

class Solution {
    public int[] solution(int N, int[] stages) {
        double[] failRate = new double[N];
        int failSum = 0;
        
        for(int i=0; i<N; i++){
            int failStage = i+1;
            int failCnt = 0;
            
            for(int j=0; j<stages.length; j++){
                if(stages[j] == failStage){
                    failCnt++;
                }
            }
            
            double rate;
            if(stages.length - failSum == 0){
                rate = 0;
            }
            else{
                rate = (double) failCnt / (double) (stages.length - failSum);
            }
            
            failRate[i] = rate;
            failSum += failCnt;
        }
        
        Integer[] order = new Integer[N];
        
        for(int i=0; i<N; i++){
            order[i] = i+1;
        }
        
        Arrays.sort(order, (a,b)->{
            if(failRate[a-1] == failRate[b-1]){
                return a-b;
            }
            return Double.compare(failRate[b-1], failRate[a-1]);
        });
        
        int[] result = new int[N];
        
        for(int i=0; i<N; i++){
            result[i] = order[i];
        }
        
        return result;
    }
}

File: Python/프로그래머스/1/42889. 실패율/실패율.py
def solution(N, stages):
    count = [0] * (N + 2)

    for stage in stages:
        count[stage] += 1

    result = []
    total = len(stages)

    for i in range(1, N + 1):
        if total == 0:
            failure = 0
        else:
            failure = count[i] / total

        result.append((i, failure))
        total -= count[i]

    result.sort(key=lambda x: (-x[1], x[0]))

    return [x[0] for x in result]